ui/app/rif/components/subDomains.js

```diff
@@ -0,0 +1,155 @@
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import rifActions from '../actions';
+import niftyActions from '../../actions';
+import {CustomButton, GenericTable} from './index';
+import ItemWithActions from './item-with-actions';
+import {SVG_PLUS} from '../constants';
+import AddNewSubdomain from '../pages/domainsDetailPage/domainDetailActive/addNewSubdomain';
+
+class Subdomains extends Component {
+
+  static propTypes = {
+    pageName: PropTypes.string.isRequired,
+    redirectParams: PropTypes.any.isRequired,
+    domainInfo: PropTypes.object,
+    isOwner: PropTypes.bool,
+    getSubdomains: PropTypes.func,
+    subdomains: PropTypes.array,
+    showPopup: PropTypes.func,
+    createSubdomain: PropTypes.func,
+    waitForListener: PropTypes.func,
+    showToast: PropTypes.func,
+    showTransactionConfirmPage: PropTypes.func,
+    isSubdomainAvailable: PropTypes.func,
+    deleteSubdomain: PropTypes.func,
+    newSubdomains: PropTypes.array,
+    paginationSize: PropTypes.number,
+    classes: PropTypes.any,
+  }
+
+  constructor (props) {
+    super(props);
+    this.state = {
+      subdomains: [],
+      addSubdomain: false,
+    };
+  }
+
+  componentDidMount () {
+    this.loadSubdomains();
+  }
+
+  componentDidUpdate (prevProps, prevState) {
+    if (prevProps.newSubdomains !== this.props.newSubdomains && this.props.newSubdomains.length > 0) {
+      this.setState({subdomains: this.props.newSubdomains});
+    }
+  }
+
+  loadSubdomains () {
+    this.props.getSubdomains(this.props.domainInfo.domainName)
+      .then(subdomains => {
+        this.setState({subdomains: subdomains});
+      });
+  }
+
+  getData () {
+  const { classes } = this.props;
+  if (this.state.subdomains) {
+      return this.state.subdomains.map((subdomain) => {
+        const item = <ItemWithActions contentClasses={classes.content} actionClasses={classes.contentActions} text={subdomain.name} enableRightChevron={true} />
```

Comment:
Same thing here you should always use parenthesis for this.

Reply:
Yeah, you can do this, the jsx sintax will take where you open the tag, and finish it. <tag></tag>... but we can use the parenthesis as a convention. I changed this in next commit